### PR TITLE
Portfolio page

### PR DESCRIPTION
**Pull request by Maks6831**
created fully functioning portfolio page
Also started loading divs

### CODE REVIEW

File: src/app/GlobalRedux/Features/CoinDatePrice/coinDateSlice.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        builder
+      .addCase(coinDatePrice.pending, (state, action) => {
+        state.loading = true;
+        state.error = "";
+      })
+      .addCase(coinDatePrice.fulfilled, (state, action) => {
+        state.loading = false;
+        state.error='';
+        if(typeof action.payload !== 'boolean'){
+          const indexIfObjectisEdit = state.data.findIndex(obj => obj.uid === action.payload.uid)
```
Comment (yoloOnTheBattlefield) [inline src/app/GlobalRedux/Features/CoinDatePrice/coinDateSlice.ts]: indexIfObjectIsEdit

File: src/app/Utils/numberFormatter.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  value: number,
+  isPercent: boolean,
+  currency: string
+): string => {
+  const formattedValue = Math.abs(value).toFixed(2);
+  const typeCheckedValue = isPercent
+    ? formattedValue + "%"
+    : parseFloat(formattedValue) > 10
+    ? currency + Math.round(parseFloat(formattedValue)).toLocaleString()
+    : currency + formattedValue;
```
Comment (yoloOnTheBattlefield) [inline src/app/Utils/numberFormatter.tsx]: find another way

File: src/app/components/AssetCard/index.tsx [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+  useEffect(() => {
+    if (portfolioData.length >= array.length) {
+      const currenMarketCoin =
+        portfolioData.find((el) => el.id === id) || coinPage;
+      setCoin(currenMarketCoin);
+    }
+  }, [portfolioData]);
+
+  useEffect(() => {
+    coin !== coinPage ? setIsLoading(true) : null;
```
Comment (yoloOnTheBattlefield) [inline src/app/components/AssetCard/index.tsx]: coin !== coinPage && setIsLoading(true)

File: src/app/components/Searchbar/index.tsx [hunk 4]
```diff
@@ -44,30 +49,42 @@ export const Searchbar = ({
...
   const setValue = () => {
     const coin = data[focusedIndex];
-    const index = coins.findIndex((obj) => obj.name === defaultValue);
-    let coinArray = [...coins];
-    coinArray[index] = coin;
-    dispatch(changeArray(coinArray));
-    setSearchInput(coin.name);
+    if (!isSearch && !isPortfolio) {
+      const index = coins.findIndex((obj) => obj.name === defaultValue);
+      let coinArray = [...coins];
```
Comment (yoloOnTheBattlefield) [inline src/app/components/Searchbar/index.tsx]: this can be const

File: src/app/page.tsx [hunk 5]
```diff
@@ -12,16 +12,18 @@ import { useAppDispatch, useAppSelector } from "./GlobalRedux/hooks";
...
+    useAppSelector((state) => state.priceChart);
   const { currency, symbol } = useAppSelector((state) => state.currencyReducer);
   const { coin } = useAppSelector((state) => state.coinReducer);
   const dispatch = useAppDispatch();
   const [currentPrice, setCurrentPrice] = useState(0);
+  const displayGraphData =
+    prices && labels && labelsTwo && market_caps && !loading;
 
   const setCoin = (value: boolean) => {
     value ? setisCoin(true) : setisCoin(false);
```
Comment (yoloOnTheBattlefield) [inline src/app/page.tsx]: setisCoin(value)

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

File: src/app/Utils/numberFormatter.tsx (hunk 2 above)
Comment (Maks6831) [reply src/app/Utils/numberFormatter.tsx]: done

**Pull request merged**